### ISSUE TITLE
[Feature request] Allow the path (or URL) for the config file to be passed as a parameter

### ISSUE THREAD

**Issue by andrewandante** [squizlabs/PHP_CodeSniffer#2831]
I've been trying to work on a solution where a collection of repos all conform to the same PHPCS config, without having to make sure that they each update their `phpcs.xml` individually whenever there is a change. It would be grand if I could run

```
phpcs --config-url https://raw.githubusercontent.com/org/repo/master/phpcs.xml
```

In my CI. Would also be useful for people wanting to do two runs on the one repo, for example:

```
phpcs --config-path ./phpcs/php5compat.xml; phpcs --config-path ./phpcs/php7compat.xml
```

Would mean you could have two (or more) complex config files in the one location without trying to pass them all in as CLI args

**Comment by jrfnl** [comment]
@andrewandante You can already set the config to a path using `--standard=/path/to/phpcs.xml`, if that helps.

Alternatively, and possibly easiest from a maintenance perspective, why not set up an external standard and point the repos all to that ?

Presuming you are using Composer, that would involve:
1. Set up a new repo with a `YourStandardName` subdirectory (choose your own name for the standard) containing just a `ruleset.xml` file with the rules you want the repos to use. The `ruleset.xml` should contain all `<rule ...>` directives, but no `<file>` or `<arg>` directives to make it project independent.
2. Add a `composer.json` file to this repo, make sure it has `"type": "phpcodesniffer-standard"` and  add `"squizlabs/php_codesniffer": "^3.5"` (or whatever version you want as minimum) in the `require` section and register the package with Packagist.
3. Now for the individual repos:
    1. Add the new package to their `require-dev`.
    2. Add the [PHPCS Composer plugin](https://github.com/Dealerdirect/phpcodesniffer-composer-installer) to `require-dev`: `"dealerdirect/phpcodesniffer-composer-installer": "^0.5"`.
        This plugin will sort out the `installed_paths` for PHPCS automatically on each `composer install`.
        You could even choose to add this plugin to the `require` of the standard instead of to the individual repo's, both will work.
    3. If the repo has a dependency on PHPCS, remove it as that dependency is now managed via the standard you created.
    4. Let each have a minimum `phpcs.xml.dist` file containing just something like this:
    ```xml
    <?xml version="1.0"?>
    <ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        name="Project X"
        xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">

        <description>Project X rules for PHP_CodeSniffer</description>

        <!--
        #############################################################################
        COMMAND LINE ARGUMENTS
        https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml
        #############################################################################
        -->

        <file>.</file>

        <!-- Only check PHP files. -->
        <arg name="extensions" value="php"/>
        <!-- Show progress, show the error codes for each message (source). -->
        <arg value="ps"/>
        <!-- Strip the filepaths down to the relevant bit. -->
        <arg name="basepath" value="./"/>
        <!-- Check up to 8 files simultaneously. -->
        <arg name="parallel" value="8"/>

        <!--
        #############################################################################
        USE THE YourStandardName RULESET
        #############################################################################
        -->

        <rule ref="YourStandardName"/>

    </ruleset>
    ```
    5. Run `composer install` on the project repo.
    6. Run `vendor/bin/phpcs` and celebrate ;-)

Any ruleset changes you want to make for all repos, you'd make to the `ruleset.xml` file in the future.
Any project repo specific changes, you can then still make in the project `phpcs.xml.dist` file.

Does that help ?

**Comment by andrewandante** [comment]
That does make a lot of sense, thanks for the detailed explanation. I think my dreams of having a "core" company repo are a bit far-reaching anyway - and if I get super-desperate, I can always WGET or CURL before I run the sniffer. Thanks!
